keps/sig-cluster-lifecycle/addons/20191013-install-addons-via-kubeadm.md

```diff
@@ -0,0 +1,450 @@
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website -->
+
+## Summary
+
+<!-- The `Summary` section is incredibly important for producing high quality user-focused documentation such as release notes or a development roadmap.
+It should be possible to collect this information before implementation begins in order to avoid requiring implementors to split their attention between writing release notes and implementing the feature itself.
+KEP editors, SIG Docs, and SIG PM should help to ensure that the tone and content of the `Summary` section is useful for a wide audience.
+
+A good summary is probably at least a paragraph in length. -->
+
+The addons subproject hosted by SIG Cluster Lifecycle has been working on implementation details regarding addon installation, user-experience, maintenance, and operation.
+The [addon-installer library POC](https://github.com/kubernetes-sigs/addon-operators/pull/25) functions as a small, vendorable implementation for applying addons to the cluster using kubectl and Kustomize. The `AddonInstallerConfiguration` API kind is intended to be a user-facing ComponentConfig.
+This KEP proposes an additional phase to kubeadm that invokes the addon-installer library.
+
+## Motivation
+
+<!-- This section is for explicitly listing the motivation, goals and non-goals of this KEP.
+Describe why the change is important and the benefits to users.
+The motivation section can optionally provide links to [experience reports][] to demonstrate the interest in a KEP within the wider Kubernetes community. -->
+
+Kubeadm currently installs two "core" addons into the cluster once the control-plane is running:  kube-proxy and CoreDNS.
+Optionally, it also supports installing kube-dns in place of CoreDNS.
+Installation of these addons occurs via application of manifests for which the strings are built into and versioned with the kubeadm binary.
+It is not currently possible to disable installation of these addons or modify them with a formal kubeadm init.
+
+Upgrades are accomplished with modifications to kubeadm's upgrade logic, often by introspecting which GVK's exist in the current cluster. 
+
+Kubeadm installs these "core" addons leaving an almost functioning cluster available for the user, but a CNI implementation is not installed. That is left to the cluster operator.
+
+Maintaining these manifests and this logic with kubeadm has to this point has allowed users to recieve a useful, default setup.
+As the ecosystem grows, this is proving to now be too inflexible for users and vendors.
+
+Use of ClusterAPI means less operators will be manually mutating clusters before they are expected to function.
+Kubeadm does not need to maintain CNI installation, but users should be able to use kubeadm to produce a functioning cluster, complete with with operational CNI. 
```

Comment:
Can we say something about creating a level playing-field for 3rd-party addons beside the set that Kubernetes/kubeadm maintainers have decided to include?

Reply:
@dholbach added a mention -- thanks